### PR TITLE
Remove native implementation of the v6 (nv14) actors.

### LINKED ISSUE #1552: Remove native implementation of v6 actors.
**Issue summary**

We no longer support nv14 and should remove our copy of the v6 actors. The code is highly intertwined so it's not trivial to untangle.

**Other information and links**

If we decide to support nv14 in the future, we'll do it through the FVM rather than native actors.

### PR DESCRIPTION
**Pull request by lemmih**
**Summary of changes**
Changes introduced in this pull request:
- Removed the `DefaultRuntime`.
- Removed the native v6 actor implementation.

**Reference issue to close (if applicable)**

Closes #1552

**Other information and links**

### CODE REVIEW

File: vm/interpreter/src/gas_tracker/mod.rs [hunk 1]
```diff
@@ -8,6 +8,7 @@ pub use self::gas_charge::GasCharge;
 pub use self::price_list::{price_list_by_epoch, PriceList};
 use vm::{actor_error, ActorError, ExitCode};
 
+// FIXME: Use gas tracker from the FVM.
```
Comment (LesnyRumcajs) [inline vm/interpreter/src/gas_tracker/mod.rs:11]: is there an issue for that?

Comment (lemmih) [reply vm/interpreter/src/gas_tracker/mod.rs:11]: There is now.

File: vm/actor_interface/src/adt/map.rs [hunk 2]
```diff
@@ -17,7 +18,8 @@ pub enum Map<'a, BS, V> {
     // V3(actorv3::Map<'a, BS, V>),
     // V4(actorv4::Map<'a, BS, V>),
     // V5(actorv5::Map<'a, BS, V>),
-    V6(actorv6::Map<'a, BS, V>),
+    // V6(actorv6::Map<'a, BS, V>),
+    V7(fil_actors_runtime_v7::fvm_ipld_hamt::Hamt<FvmRefStore<'a, BS>, V, BytesKey>),
```
[the remark this answers is not in the document]
Comment (lemmih) [reply vm/actor_interface/src/adt/map.rs:22]: I'll remove it.

File: utils/statediff/src/lib.rs [hunk 3]
```diff
@@ -115,22 +115,23 @@ fn pp_actor_state(
...
-    } else if let Ok(account_state) = ipld::from_ipld::<account::State>(ipld.clone()) {
-        write!(&mut buffer, "{:?}", account_state)?;
-    } else if let Ok(state) = ipld::from_ipld::<power::State>(ipld.clone()) {
-        write!(&mut buffer, "{:?}", state)?;
-    } else if let Ok(state) = ipld::from_ipld::<market::State>(ipld.clone()) {
-        write!(&mut buffer, "{:?}", state)?;
-    } else {
-        buffer += &serde_json::to_string_pretty(&resolved)?;
-    }
+    // FIXME: Use the actor interface to load and pretty print the actor states.
```
[the remark this answers is not in the document]
Comment (lemmih) [reply utils/statediff/src/lib.rs:123]: We do now.

File: vm/actor_interface/Cargo.toml [hunk 4]
```diff
@@ -10,7 +10,7 @@ edition = "2021"
 #actorv3 = { package = "forest_actor", version = "2.0" }
 #actorv4 = { package = "forest_actor", version = "3.0" }
 #actorv5 = { package = "forest_actor", version = "4.0" }
-actorv6 = { package = "forest_actor", path = "../actor" }
+#actorv6 = { package = "forest_actor", path = "../actor" }
```
[the remark this answers is not in the document]
Comment (lemmih) [reply vm/actor_interface/Cargo.toml]: Removed.

**Review by LesnyRumcajs — APPROVED** [review]

**Review by connormullett — APPROVED** [review]
lgtm

**Pull request merged**